test/resource_test.exs

```diff
@@ -24,18 +24,29 @@ defmodule ResourceTest do
           language_id: "25",
           modes: ["USSD", "TEXT"],
           value: "non-existent language"
+        },
+        %ResourceValue{
+          language_id: "22",
+          modes: ["USSD"],
+          value: "english"
         }
       ]
     }
 
     # no matching language
-    assert {:error, _} = Resource.matching_resource(resource, "eng", "TEXT", @flow)
+    assert Resource.matching_resource(resource, "eng", "TEXT", @flow) ==
+             {:error, "no matching resource"}
 
     # no matching mode.
-    assert {:error, _} = Resource.matching_resource(resource, "fra", "RICH_MESSAGING", @flow)
+    assert Resource.matching_resource(resource, "fra", "RICH_MESSAGING", @flow) ==
+             {:error, "no matching resource"}
 
     # matching language and mode
     assert {:ok, %ResourceValue{value: "french"}} =
              Resource.matching_resource(resource, "fra", "TEXT", @flow)
+
+    # no matching languages for resource should use default language
+    assert {:ok, %ResourceValue{value: "english"}} =
+             Resource.matching_resource(resource, "por", "USSD", @flow)
```

Comment:
@joaothallis I'm actually not sure about this change. This could generate unexpected behaviors. If the current language is Portuguese, why would that return something that is in English?

Is there any other test case that allows this scenario?

Reply:
@santiagocardo it's returning English because it's default  ([first language in the list of flow languages](https://github.com/turnhub/flow_runner/pull/247/files#diff-822796d199d7c9b2869fe88753b8ee4acb82709e49282beec40cfb2b1fa2f8a5R10)) language